x-pack/plugins/enterprise_search/server/collectors/app_search/telemetry.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { set } from 'lodash';
+import { ISavedObjectsRepository, SavedObjectsServiceStart } from 'src/core/server';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+
+import { AS_TELEMETRY_NAME, ITelemetrySavedObject } from '../../saved_objects/app_search/telemetry';
+
+/**
+ * Register the telemetry collector
+ */
+
+interface Dependencies {
+  savedObjects: SavedObjectsServiceStart;
+  usageCollection: UsageCollectionSetup;
+}
+
+export const registerTelemetryUsageCollector = ({
```

Comment:
This example shows a check for the existence of `usageCollection` since it is an optional dependency. Do we need to do the same thing here in case `usageCollection` is not registered? Or is it checked upstream somewhere?

Reply:
For this specific line/function - no, since `registerTelemetryUsageCollector` is only called in plugin.ts (upstream) wrapped in the `if (usageCollection)` check, and should never be called anywhere else.

In general - I checked with the telemetry folks on Slack (you can check the channel) and their answer has been that we don't need to do any extra checks on our end or on the client side etc - their usageCollection plugin should be managing most of that for us.